test/lit_tests/canonicalizefor/doWhile.mlir

```diff
@@ -0,0 +1,168 @@
+// RUN: enzymexlamlir-opt --allow-unregistered-dialect --canonicalize-scf-for -split-input-file %s | FileCheck %s
+module @simple{
+  func.func @do_while() -> index {
+    %c0 = arith.constant 0 : index
+    %c1 = arith.constant 1 : index
+    %c5 = arith.constant 5 : index
+
+    %result = scf.while (%i = %c0) : (index) -> index {
+      "before.keepalive"(%i) : (index) -> ()
+      %updated = arith.addi %i, %c1 : index
+      %cond = arith.cmpi slt, %updated, %c5 : index
+      scf.condition(%cond) %updated : index
+    } do {
+    ^bb0(%new_i: index):
+      scf.yield %new_i : index
+    }
+    
+    return %result : index
+  }
+}
+// CHECK-LABEL:   module @simple {
+// CHECK:           func.func @do_while() -> index {
+// CHECK:             %[[VAL_0:.*]] = arith.constant 5 : index
+// CHECK:             %[[VAL_1:.*]] = arith.constant 0 : index
+// CHECK:             %[[VAL_2:.*]] = arith.constant 1 : index
+// CHECK:             scf.for %[[VAL_3:.*]] = %[[VAL_1]] to %[[VAL_0]] step %[[VAL_2]] {
```

Comment:
for my sake I wrote this out, and it looks reasonable, but can someone (@ivanradanov or @Pangoraw double check the internal logic)?

the above code runs
```
i = 0
keepalive(i = 0)
updated = i + 1 = 1
if !(updated < 5) break

i = updated = 1
keepalive(i = 1)
updated = i + 1 = 2
if !(updated < 5) break


i = updated = 2
keepalive(i = 2)
updated = i + 1 = 3
if !(updated < 5) break

i = updated = 3
keepalive(i = 3)
updated = i + 1 = 4
if !(updated < 5) break

i = updated = 4
keepalive(i = 4)
updated = i + 1 = 5
if !(updated < 5) break;
  -> breaks
```